src/provider/controller/IntentHandler.ts

```diff
@@ -0,0 +1,74 @@
+import {injectable, inject } from "inversify";
+import {Inject} from "../common/Injectables";
+import {Intent} from "../../client/intents";
+import {SelectorHandler, SelectorResult} from "./SelectorHandler";
+import {IntentResolution, Application } from "../../client/main";
+import {ContextHandler} from "./ContextHandler";
+import {FindFilter, Model} from "../model/Model";
+import {AppDirectory} from "../model/AppDirectory";
+import {AppWindow} from "../model/AppWindow";
+
+@injectable()
+export class IntentHandler {
+    @inject(Inject.APP_DIRECTORY)   private _directory!: AppDirectory;
+    @inject(Inject.MODEL)           private _model!: Model;
+    @inject(Inject.SELECTOR)        private _selector!: SelectorHandler;
+    @inject(Inject.CONTEXT_HANDLER) private _contexts!: ContextHandler;
+
+    private _promise: Promise<IntentResolution>|null;
+
+    constructor() {
+        this._promise = null;
+    }
+
+    public async raise(intent: Intent): Promise<IntentResolution> {
```

Comment:
Maybe handled elsewhere and I just haven't seen it yet, but shouldn't this accept a target too?

Reply:
`target` would be a (optional) member of `Intent`. Thought I'd added that, but seems not.

Will add the field to `Intent`, but will save the actual implementation in the provider to a future story.

Reply:
Sure. I'd guessed this was just an oversight, but will definitely need implementing asap as it's in the spec and used but the tests/demo apps